src/NServiceBus.Core/Sagas/SimpleSaga.cs

```diff
@@ -0,0 +1,85 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// This class is used to define sagas containing data and handling a message.
+    /// To handle more message types, implement <see cref="IHandleMessages{T}" />
+    /// for the relevant types.
+    /// To signify that the receipt of a message should start this saga,
+    /// implement <see cref="IAmStartedByMessages{T}" /> for the relevant message type.
+    /// </summary>
+    /// <typeparam name="TSagaData">A type that implements <see cref="IContainSagaData" />.</typeparam>
+    public abstract class SimpleSaga<TSagaData> : Saga where TSagaData : IContainSagaData, new()
+    {
+
+        /// <summary>
+        /// Gets the name of the correlation property for <typeparamref name="TSagaData"/>.
+        /// </summary>
+        protected abstract string CorrelationPropertyName { get; }
```

Comment:
out of curiosity, would it also be possible to use the member-returning function approach as the "regular" saga mapping uses (e.g. something like `CorrelationProperty = sagaData => sagaData.CorrelationId` ) instead of a string based API?